### ISSUE TITLE
Mumakil Rework

### ISSUE THREAD

**Issue by JukeJukestar** [MattTheLegoman/RealmsInExile#475]
Foreword:
I initially made this issue under the impression that
a) Mumaks were native to nafarat and the lands south of there (my mistake) and that
b) they were not able to be recruited as MAA. This was a misunderstanding on my part, as they are avaiable to all northern apysaic rulers, but very expensive to the point that i dont believe i have ever seen them in a game exept for the initial two Suladan starts with.

# Why its good
(At least from my perspective) In the books, the Mumaks are mentioned a few times, along with the Haradrim reinforcements being ambushed in Ithilien and during the battle of pelennor fields. There are a few pages denoted to them specifically and the wonder the hobbits feel about them. They are quite iconic and very strongly connected to our view of the haradrim in the books and movies.

Their exact cost would have to be tweaked, but in general they should be rather expensive and seen less as a cost efficient machine of war (and terror) and more as a prestige object for rich rulers to add to their hosts.

EDIT: Just to give you a sense of scale for their current cost: Suladan starts with two siege mumaks, which cost half of his initial income and almost single handedly bankrupt him after the initial pezarsan war in most games. (Especially if he gets sieged and his control levels start to go down, which they very often do)

Currently, they are a very rare sight since they are incredibly costly (1.4k gold) to the point of near uselessness (unless tribal, where they could be viable?). It also does not make too much sense for only near haradrim rulers to be able to recruit them, since from my understanding they are native to parts of the map not yet implemented.

If this decision was implemented, the ability to freely recruit them for near haradrim would be removed and moved into a tradition. They would become a bit more common (in fact i dont think i have ever seen anyone recruit them until now), but they would no longer spammable since a ruler can only have a very limited ammount of them. It may also be worth looking into streamlining them since Siege Mumaks are not explained in game (Despite the game not showing us, I would assume they work as siege engines?)

# Decision requirements:
- Only avaiable to Haradrim rulers of king tier or higher (convincing the Riders to join you and actually feeding the Mumaks would be ressource intensive so it shouldnt be avaiable to dukes)
- Only avaiable to independent rulers
- 5-10 year cooldown, only useable if you dont already have a host of War Mumak
- Eventually, once their native teritory along with a Mumak Rider tradition is implemented, only avaiable if there are independent rulers with the tradition or counties with the tradition inside your realm. (Since if they were conquered by foreign powers such as the Golden King, they would seek to use them for their own hosts)

# Actual decision rundown
- Rulers would be able to decide from three price offers to send (Lowballing, good offer, high cost) via event and send an envoy to a group of Mumak Riders
- Depending on their offer as well as diplomacy skill (maybe even cultural acceptance could play a role?), the chance of the riders accepting would be different (with an almost guarantueed acceptance at a high offer)
- If successful, the ruler gains a special event host with a small host of Mumaks, as well as some prestige and court grandeur (since they are mostly a prestige object that could even be seen as a money sink for rich kingdoms like Rây, and not as a cost efficent troop investment if you have to import them)
- Unit reinforces, but costs upkeep and disbands on succession
- If not successful, some prestige and court grandeur is lost and the money is not taken.

If this is too complicated, perhaps the decision could be streamlined by removing either the "gold offer" or "size" choice as well as acceptance chance (although i really like the idea of character diplomacy factoring in as a skill check)

**Comment by Remerod** [comment]
There have been some talk about remaking a better innovations for Mumakils which would also add a duchy building improving them, more information to come

**Comment by Remerod** [comment]
There will be a Mumakil MAA change, it not be the one that's been suggested here but i will rename this issue and move it to non-release critical. 

If @VectorMaximus want to specify what's he've planned up for them he is free to elaborate it here :)

**Comment by jj248** [comment]
@VectorMaximus - what's the status of this?  Ready to test?

**Comment by jj248** [comment]
Is this now in a testable state @VectorMaximus ?

**Comment by VectorMaximus** [comment]
Updated Mumakil Recruitment System:

Haradrim Cultures gain access to the 'Mumakil Domestication' Innovation (not all start with it unlocked, but available to all in the Apysaic heritages). This unlocks two things:
1. Constructing 'Mumakil Pen' ducal buildings in appropriate regions. If you have one in your demense and the appropriate innovation, you can recruit Mumakil as MAA. This recruitment can also be done if you have the level 3 Palace of Haruzan.
2. You can take the decision to import Mumakil from the south. This will fire an event that allows you to barter for Mumakil at increased prices compared to doing it through MAA.

Key things for feedback:
1. Does the system work? IE can you only recruit Mumakil as described above, or are they still getting through in other forms?
2. Does the system feel appropriate and balanced? Are the MAA worth the investment?

Ready for testing.

**Comment by Remerod** [comment]
I think it works? although should there be a difference if you buy them from outside the "lands of the Mumakil" or from inside? because there is none currently, it cost the same whatever you'r in Ray or haruzan (i'm using Ray as an other example as the Bôzishnarûd also has the 'Mumakil Domestication' Innovation and are inside the "lands of the Mumakil"), but i guess this is whatever if it supposed to be that way

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/79544635/219762763-30e2faf2-cd46-483d-8a4f-79673e518d62.png)
pursing from within Haruzan

![image](https://user-images.githubusercontent.com/79544635/219762982-32ad6824-6881-44d6-af0e-017d6c0f575c.png)
pursing from within Ray

![image](https://user-images.githubusercontent.com/79544635/219763418-9a847ade-5b17-4b19-a508-9827a4d63a2a.png)
the decision makes it sound as if there is a difference if your in our out of it.

</details>

when i taken the decision as ray (and have build the first level of the ducal pen) i can't recruit them as MAA but as Haruzan i can recruit War Mumakil.  And if i go to Aegardh (haruze free duchy) i can create war mumakil MAA
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/79544635/219768005-168c2f7d-1a6a-4494-a0f7-ae3084495cc1.png)
having the building as the king of ray in another duchy(that he owns and controls but not the main) but unable to build the maa

![image](https://user-images.githubusercontent.com/79544635/219769387-adcb28be-7627-47a7-aa20-29f0971f83e0.png)
Aegardh not having taken the decision

![image](https://user-images.githubusercontent.com/79544635/219769487-601a3edf-8c6e-434d-bcdd-6332c734f415.png)
but being able to recruit them still

</details>

> you can take the decision to import Mumakil from the south. This will fire an event that allows you to barter for Mumakil at increased prices compared to doing it through MAA.

so this seems to work, i think the cost is good to. Although when you have taken the decision gotten your army, your army have gotten killed, will you be able to take the decision again, waiting a year the decision does not seem to reappear, so is there an invisible cool down for the decision (if so i think it should be visible) and if not maybe we could make it so it reappears if you no longer have the special event regiment but have a cooldown period like 4, 5  or 10 years? or are you just suppose to build your own from the mumakils pen's now?

Also are you supposed to be able to construct the mumakil pens without taking the decision if your lord have taken it? becouse this dude can without taken the decision (although his lord Suladan has)
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/79544635/219772507-ec1d0f2f-6afd-4bdd-bcb4-668aed3f4642.png)

</details>

also it needs to be made more clear in the decision that the 'Mumakil Pen' are supposedly a ducal pen that you can build. I very much like this idea and theme but i think the information about it being a build-able ducal building is bit weak at the moment. 

So to summarize, there might be some wacky stuff with the decision and haruze being able to recruit the Mumkils + build their mumakil pens and the decision need to make it more clear that you are needing to build mumakil pens in your ducal building spots

Generally i think this is great idea though!

**Comment by VectorMaximus** [comment]
So, issues:

1. Unlocking conditions didn't have the correct scoping, so those have now been fixed. It is now correctly unlocked by the buildings.
2. It was also being unlocked by an old innovation. That has now been fixed.

On the other things:
You pay an inflated price from a trader, even if the lands of the Mumakil. 
There is a ten year cooldown for purchasing Mumakil (so you can't just infinitely stack them - they are quite strong). Agreed that such should probably be shown somehow. You don't get the cooldown if you don't purchase though (if you back out of the decision).

**Comment by Remerod** [comment]
tested it again, seems to be working fine (decision text could be a bit better) but it's fine for closing now, good work! :)

**Comment by Remerod** [comment]
tested it again, seems to be working fine (decision text could be a bit better) but it's fine for closing now, good work! :)
